### PR TITLE
add version_added to TaxonomyEntity

### LINKED ISSUE #4134: 422 error when updating a taxonomy entity
### Bug Description

When updating a default data use, data subject, or data category in the taxonomy an error is thrown and the update is not saved.

#### Steps to Reproduce

1. run fides or use the rc env
2. go to the taxonomy page https://fides-plus-rc.fides-staging.ethyca.com/taxonomy
3. hover over an entity 
4. click the pencil 
5. make an edit 
6. click Update entity 
7. see the error 

### Expected behavior

Updates should save without error 

### Screenshots

image

### Environment

* Version: 2.20.1

### PR DESCRIPTION
**Pull request by Kelsey-Ethyca**
Closes #4134 

### Description Of Changes

add version_added to TaxonomyEntity

### Steps to Confirm

- [ ] run fides 
- [ ] go to the taxonomy page
- [ ] hover over an entity
- [ ] click the pencil
- [ ] make an edit
- [ ] click Update entity
- [ ] verify there is no error and the update is saved 

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: clients/admin-ui/src/types/api/models/DataCategory.ts [hunk 1]
```diff
@@ -32,4 +32,8 @@ export type DataCategory = {
    * Indicates whether the resource is currently 'active'.
    */
   active?: boolean;
+  /**
+   * This is for tracking when the daulft entity was added
```
Comment (pattisdr) [inline clients/admin-ui/src/types/api/models/DataCategory.ts:36]:
```suggestion
   * This is for tracking when the default entity was added
```

File: clients/admin-ui/src/types/api/models/DataSubject.ts [hunk 2]
```diff
@@ -47,4 +47,8 @@ export type DataSubject = {
    * Indicates whether the resource is currently 'active'.
    */
   active?: boolean;
+  /**
+   * This is for tracking when the daulft entity was added
```
Comment (pattisdr) [inline clients/admin-ui/src/types/api/models/DataSubject.ts:51]:
```suggestion
   * This is for tracking when the default entity was added
```

File: clients/admin-ui/src/types/api/models/DataUse.ts [hunk 3]
```diff
@@ -55,4 +55,8 @@ export type DataUse = {
    * Indicates whether the resource is currently 'active'.
    */
   active?: boolean;
+  /**
+   * This is for tracking when the daulft entity was added
```
Comment (pattisdr) [inline clients/admin-ui/src/types/api/models/DataUse.ts:59]:
```suggestion
   * This is for tracking when the default entity was added
```

File: clients/admin-ui/src/features/taxonomy/hooks.tsx [hunk 4]
```diff
@@ -78,6 +78,7 @@ const transformTaxonomyBaseToInitialValues = (
   description: t.description ?? "",
   parent_key: t.parent_key ?? "",
   is_default: t.is_default ?? false,
+  version_added: t.version_added ?? "",
```
Comment (pattisdr) [inline clients/admin-ui/src/features/taxonomy/hooks.tsx:81]: Looking more closely at the db here, I think we should need to write back the values of `version_deprecated` and `replaced_by`, or we could end up overwriting those if they are populated.

**Review by pattisdr — COMMENTED** [review]
Tested this works well nice find @Kelsey-Ethyca! I just have a comment about adding additional fields that aren't throwing errors but we would also be writing back.

- Tested updating default uses, categories, and subjects, as well as creating/updating uses, categories and subjects.


@ThomasLaPiana FYI this validation over in fideslang seems to now be requiring this field in the request body when updating. In general, it seems like it would be useful if the CTL endpoints ignored fields that were not in the request body.  We had a similar thing last week, where we realized fields not in the System request were getting set to null https://github.com/ethyca/fides/pull/4116.  What are your thoughts on that approach?

**Review by pattisdr — APPROVED** [review]
great @Kelsey-Ethyca thank you

**Comment by ThomasLaPiana** [comment]
> Tested this works well nice find @Kelsey-Ethyca! I just have a comment about adding additional fields that aren't throwing errors but we would also be writing back.
> 
> * Tested updating default uses, categories, and subjects, as well as creating/updating uses, categories and subjects.
> 
> @ThomasLaPiana FYI this validation over in fideslang seems to now be requiring this field in the request body when updating. In general, it seems like it would be useful if the CTL endpoints ignored fields that were not in the request body. We had a similar thing last week, where we realized fields not in the System request were getting set to null #4116. What are your thoughts on that approach?

Sounds logical to me! I think that would be a pretty easy fix

**Pull request merged**